### ISSUE TITLE
Program March 2024, Challenge 2

### ISSUE THREAD

**Issue by Mesure73L** [twelve-of-code-official/challenges#4]
> [!WARNING]
> Spoiler warning, we are talking about the specific challenges in this issue.

**Comment by Mesure73L** [comment]
I have this code to get the user's starting brand:
```js
const startingBrand =
    finalBrandNames[
        Math.floor(smanRandom2024mar1.startingBrand * finalBrandNames.length)
    ];
```
It is returning `undefined`. How can I fix this?

**Comment by Mesure73L** [comment]
@BeepBot99

**Comment by BeepBot99** [comment]
Please don't mention me in issues that I am subscribed to (every issue I have commented in).
Please debug things on your own first. Don't ask me things like this that are simple.
I believe that finalBrandsName doesn't exist in the global scope.

**Comment by BeepBot99** [comment]
Also check that the index exists. Please don't ask for help with simple things like this. You're a person, you can do things.

**Comment by Mesure73L** [comment]
thanks for your positive help ig /s

**Comment by Mesure73L** [comment]
Can you send me the php code I need for sman and the logic for the answer checker? I am a bit stuck on what I need to do for it. Thanks!

**Comment by BeepBot99** [comment]
The PHP code for `sman` is in https://github.com/twelve-of-code-official/website/issues/173. For the answer checker, you just need to run the same code that the user would run and the code that the API would run and check if you get the same answer. However, you should run the API's code in the JavaScript (client side) so that it doesn't have to wait for the API to respond.

**Comment by Mesure73L** [comment]
This is my current plan (in order of what I'm planning on completing it in). Is this good, and is there anything missing?
![image](https://github.com/twelve-of-code-official/challenges/assets/115181664/a25b15e8-84aa-4341-9ef2-4736d1e2b69d)

**Comment by BeepBot99** [comment]
Ok, so that is mostly good, except I wouldn't hard code test seed. I would just make it 0 or something simple like that. Currently, 0 is not allowed as a seed, so you can't just enter 0 as your real seed.

**Comment by Mesure73L** [comment]
> Ok, so that is mostly good, except I wouldn't hard code test seed. I would just make it 0 or something simple like that. Currently, 0 is not allowed as a seed, so you can't just enter 0 as your real seed.

We already discussed the seed and agreed on it being `TEST_SEED`.

**Comment by BeepBot99** [comment]
No, I mean, instead of preprogramming every single possibility for test seed, just when the API gets sent it, it acts as if it were 0.

**Comment by Mesure73L** [comment]
It's not that many, i'll just hard code it.

**Comment by BeepBot99** [comment]
No, please don't.

**Comment by BeepBot99** [comment]
What do you mean it's not that many, it's 129.

**Comment by Mesure73L** [comment]
Do you even know what `TEST_SEED` is?

**Comment by Mesure73L** [comment]
Because it's a lot less than 129.

**Comment by Mesure73L** [comment]
I literally put that in the challenge for that reason.

**Comment by BeepBot99** [comment]
> No, `TEST_SEED` is made for you to use **in any challenge where you need to input your seed** so that you can use the example in your program. The example is the same for every person. `TEST_SEED` is for the very simple example that was given on the challenge for testing purposes. If `TEST_SEED` is not hardcoded, it will have to **always** give the **same** response as what was given in the challenge example. Final decision.

Yeah, any seed always gives the same response 🤯.

**Comment by BeepBot99** [comment]
I'm saying if `TEST_SEED` is entered as the seed in the API, then it should just act as if it were the seed 0 but with a limited dataset.

**Comment by BeepBot99** [comment]
I'm not letting you hardcode every single possibility.

**Comment by Mesure73L** [comment]
If you want to do that for your challenges, go ahead. But for my challenges, I'm doing it this way because it's easier for me. You haven't listed any downsides yet. Plus I already did and it doesn't make the code worse at all.

**Comment by Mesure73L** [comment]
> Yeah, any seed always gives the same response 🤯.

No, I mean the same response as is already on the challenge HTML.

**Comment by Mesure73L** [comment]
# 🤯

**Comment by BeepBot99** [comment]
Ok so clearly you don't know what like the whole point of a RESTful API is.

**Comment by Mesure73L** [comment]
You can use whatever code you want in your challenge as long as it works as expected. Unlike you, I don't care about how the code was written, as long as it works correctly.
